app/moves/middleware.js

```diff
@@ -116,24 +115,52 @@ const movesMiddleware = {
           return moveService
             .getMovesCount({ dateRange, status: moveType.filter, locationId })
             .then(count => {
-              const { label } = moveType
               return {
-                label,
+                ...moveType,
                 value: count,
                 active: moveType.filter === req.params.status,
                 href: `${req.baseUrl}/${period}/${date}${
                   locationId ? '/' + locationId : ''
                 }/${moveType.filter}`,
               }
             })
-            .then(presenters.moveTypesToFilterComponent)
         })
       )
       next()
     } catch (error) {
       next(error)
     }
   },
+  setTotalMoves: (req, res, next) => {
+    if (!get(res, 'locals.moveTypeNavigation')) {
```

Comment:
Also, if its in the controller you can remove logic like this as you can assume it has the correct middleware assigned, otherwise we do want it to blow up and error.